### ISSUE TITLE
SSH clients is not handling connections well

### ISSUE THREAD

**Issue by mnaiman** [duplicati/duplicati#1800]
From log of my SFTP server can be seen how number of sessions is increasing with each file. At the end there are hundreds of them.

```
ACCEPT|09.06.2016 22:26:44|There are now 3 active connections
ACCEPT|09.06.2016 22:26:44|There are now 4 active connections
ACCEPT|09.06.2016 22:26:44|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:44|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
LIST|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *150 LIST /backup/Backup_Name/*
LIST|09.06.2016 22:26:44|[SFTP:2406720_0:username:IP_ADDRESS] READ: *226 LIST /backup/Backup_Name/*
ACCEPT|09.06.2016 22:26:47|[SFTP:lookup:55556][2406721] Accepting connection from: IP_ADDRESS:56371
ACCEPT|09.06.2016 22:26:47|There are now 5 active connections
ACCEPT|09.06.2016 22:26:47|There are now 6 active connections
ACCEPT|09.06.2016 22:26:47|There are now 5 active connections
ACCEPT|09.06.2016 22:26:47|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:47|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes*
MDTM|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *fileExists /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes:true*
RETR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *RETR START /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes*
RETR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *RETR /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes:dyrHS*
RETR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] WROTE: *150 Opening BINARY mode data connection for /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes (2205 bytes). ("/backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes") R E T R*
RETR|09.06.2016 22:26:47|226-Download File Size:2205 bytes @ 2K/sec.
RETR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] WROTE: *226 Transfer complete. MD5=36e2bdc8f62716841fa41f28a9c4cfdf ("/backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes") RETR*
RETR|09.06.2016 22:26:47|[2406721:username:IP_ADDRESS] WROTE: *Adding Backup_Name-20150427T033000Z.dlist.zip.aes to event tracking.*
RETR|09.06.2016 22:26:47|[SFTP:2406721_0:username:IP_ADDRESS] READ: *RETR END   /backup/Backup_Name/Backup_Name-20150427T033000Z.dlist.zip.aes:dyrHS*
ACCEPT|09.06.2016 22:26:48|[SFTP:lookup:55556][2406722] Accepting connection from: IP_ADDRESS:56372
ACCEPT|09.06.2016 22:26:48|There are now 6 active connections
ACCEPT|09.06.2016 22:26:48|There are now 7 active connections
ACCEPT|09.06.2016 22:26:48|There are now 6 active connections
ACCEPT|09.06.2016 22:26:48|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:48|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes*
MDTM|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *fileExists /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes:true*
RETR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *RETR START /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes*
RETR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *RETR /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes:CtE6M*
RETR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] WROTE: *150 Opening BINARY mode data connection for /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes (2205 bytes). ("/backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes") R E T R*
RETR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] READ: *RETR END   /backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes:CtE6M*
RETR|09.06.2016 22:26:48|226-Download File Size:2205 bytes @ 2K/sec.
RETR|09.06.2016 22:26:48|[SFTP:2406722_0:username:IP_ADDRESS] WROTE: *226 Transfer complete. MD5=91c9abb61dc3815ace79411b5a75ec3f ("/backup/Backup_Name/Backup_Name-20150428T033000Z.dlist.zip.aes") RETR*
RETR|09.06.2016 22:26:48|[2406722:username:IP_ADDRESS] WROTE: *Adding Backup_Name-20150428T033000Z.dlist.zip.aes to event tracking.*
ACCEPT|09.06.2016 22:26:48|[SFTP:lookup:55556][2406723] Accepting connection from: IP_ADDRESS:56373
ACCEPT|09.06.2016 22:26:48|There are now 7 active connections
ACCEPT|09.06.2016 22:26:48|There are now 8 active connections
ACCEPT|09.06.2016 22:26:48|There are now 7 active connections
ACCEPT|09.06.2016 22:26:48|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:48|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes*
MDTM|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *fileExists /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes:true*
RETR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *RETR START /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes*
RETR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *RETR /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes:Gb5K0*
RETR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] WROTE: *150 Opening BINARY mode data connection for /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes (2205 bytes). ("/backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes") R E T R*
RETR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] READ: *RETR END   /backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes:Gb5K0*
RETR|09.06.2016 22:26:48|226-Download File Size:2205 bytes @ 2K/sec.
RETR|09.06.2016 22:26:48|[SFTP:2406723_0:username:IP_ADDRESS] WROTE: *226 Transfer complete. MD5=729cef272df62a41d9739f3e570540f7 ("/backup/Backup_Name/Backup_Name-20150429T033000Z.dlist.zip.aes") RETR*
RETR|09.06.2016 22:26:48|[2406723:username:IP_ADDRESS] WROTE: *Adding Backup_Name-20150429T033000Z.dlist.zip.aes to event tracking.*
ACCEPT|09.06.2016 22:26:49|[SFTP:lookup:55556][2406724] Accepting connection from: IP_ADDRESS:56374
ACCEPT|09.06.2016 22:26:49|There are now 8 active connections
ACCEPT|09.06.2016 22:26:49|There are now 9 active connections
ACCEPT|09.06.2016 22:26:49|There are now 8 active connections
ACCEPT|09.06.2016 22:26:49|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:49|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes*
MDTM|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *fileExists /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes:true*
RETR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *RETR START /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes*
RETR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *RETR /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes:9jxvm*
RETR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] WROTE: *150 Opening BINARY mode data connection for /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes (2205 bytes). ("/backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes") R E T R*
RETR|09.06.2016 22:26:49|226-Download File Size:2205 bytes @ 2K/sec.
RETR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] WROTE: *226 Transfer complete. MD5=2ad8fc49bee0dfeba79ff61206db5ad3 ("/backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes") RETR*
RETR|09.06.2016 22:26:49|[2406724:username:IP_ADDRESS] WROTE: *Adding Backup_Name-20150430T033000Z.dlist.zip.aes to event tracking.*
RETR|09.06.2016 22:26:49|[SFTP:2406724_0:username:IP_ADDRESS] READ: *RETR END   /backup/Backup_Name/Backup_Name-20150430T033000Z.dlist.zip.aes:9jxvm*
ACCEPT|09.06.2016 22:26:49|[SFTP:lookup:55556][2406725] Accepting connection from: IP_ADDRESS:56375
ACCEPT|09.06.2016 22:26:49|There are now 9 active connections
ACCEPT|09.06.2016 22:26:49|There are now 10 active connections
ACCEPT|09.06.2016 22:26:49|There are now 9 active connections
ACCEPT|09.06.2016 22:26:49|Client is attempting none authentication
ACCEPT|09.06.2016 22:26:49|Client is attempting password authentication
ACCEPT|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *Verifying password for username.*
USER|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *USER username*
USER|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *CONNECT username*
MDTM|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/*
MDTM|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/*
CWD|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *CWD /backup/Backup_Name/*
STOR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] WROTE: *250 "/Backup_Name/" CWD command successful.*
MDTM|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *getFileAttributes /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes*
MDTM|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *fileExists /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes:true*
RETR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *RETR START /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes*
RETR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *RETR /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes:nyS5U*
RETR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] WROTE: *150 Opening BINARY mode data connection for /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes (2205 bytes). ("/backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes") R E T R*
RETR|09.06.2016 22:26:49|226-Download File Size:2205 bytes @ 2K/sec.
RETR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] WROTE: *226 Transfer complete. MD5=f12fbb323fb74f2a3b934c4c8cbad904 ("/backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes") RETR*
RETR|09.06.2016 22:26:49|[2406725:username:IP_ADDRESS] WROTE: *Adding Backup_Name-20150501T033000Z.dlist.zip.aes to event tracking.*
RETR|09.06.2016 22:26:49|[SFTP:2406725_0:username:IP_ADDRESS] READ: *RETR END   /backup/Backup_Name/Backup_Name-20150501T033000Z.dlist.zip.aes:nyS5U*
ACCEPT|09.06.2016 22:26:50|[SFTP:lookup:55556][2406726] Accepting connection from: IP_ADDRESS:56376
ACCEPT|09.06.2016 22:26:50|There are now 10 active connections
ACCEPT|09.06.2016 22:26:50|There are now 11 active connections
ACCEPT|09.06.2016 22:26:50|There are now 10 active connections
```

**Comment by mnaiman** [comment]
It think it is related to #1341

**Comment by mnaiman** [comment]
It is necessary to say, that --no-connections-reuse was used in this case.

**Comment by huste** [comment]
Does this still hapen with  2.0.1.18_canary_2016-08-10? There have been fixes for SFTP/SSH connections and it resolved my #1872

**Comment by mnaiman** [comment]
It is working
